File: cmd/list.go
package cmd

// Reads the rules file and prints the rules in a human-readable format.
import (
    "fmt"
    "github.com/spf13/cobra"
)

var listCmd = &cobra.Command{
    Use:   "list",
    Short: "Lists all current rules with their metadata",
    Run: func(cmd *cobra.Command, args []string) {
        fmt.Println("Listing all rules...")
    },
}

File: src/commands/list.rs
use crate::cli;
use crate::core::context;
use anyhow::Result;
use clap::Args;

#[derive(Args)]
#[command(about = "📋 List all current rules with their metadata")]
pub struct ListArgs;

pub fn run(_args: ListArgs) -> Result<()> {
    log::info!("Listing all rules...");

    let rf = context::get_locked_rules_file()?;
    let rules_list = rf.list_rules();

    if rules_list.is_empty() {
        cli::warning("No rules found.");
        cli::info("Use `tooka add` to create your first rule.");
        return Ok(());
    }

    cli::header(&format!("📋 Found {} rules", rules_list.len()));
    cli::rule_table_header();

    for rule in &rules_list {
        log::debug!(
            "Rule ID: {}, Name: {}, Enabled: {}",
            rule.id,
            rule.name,
            rule.enabled
        );
        cli::rule_table_row(&rule.id, &rule.name, rule.enabled);
    }

    println!();
    cli::success("Rules listed successfully!");

    Ok(())
}
